pyresample/geometry.py

```diff
@@ -2587,11 +2587,9 @@ def get_area_slices(self, area_to_cover, shape_divisible_by=None):
             return x_slice, y_slice
 
         if not self.is_geostationary:
-            raise NotImplementedError("Source projection must be 'geos' if "
-                                      "source/target projections are not "
-                                      "equal.")
-
-        data_boundary = Boundary(*get_geostationary_bounding_box_in_lonlats(self))
+            data_boundary = self.boundary(frequency=100)
+        else:
+            data_boundary = Boundary(*get_geostationary_bounding_box_in_lonlats(self))
```

Comment:
If we add `get_geostationary_bounding_box_in_lonlats` in https://github.com/mraspaud/pyresample/blob/4e245a906bf3aeff072f6b61febdd1df298fb425/pyresample/geometry.py#L299 `get_bbox_lonlats` you can simplify all the if_else conditions here and just do self.boundary() with a sensible default ...

Reply:
Here below: 

`AreaDefBoundary(area_to_cover, 100)` fails if `area_to_cover` is `SwathDefinition`.
Maybe use `area_to_cover.boundary()` also here in all cases (with sensible frequency defaults)

Reply:
I think this simplification is better suited for another PR